sql/logic_test.go

```diff
@@ -172,6 +172,20 @@ func (t *logicTest) close() {
 
 // setUser sets the DB client to the specified user, creating it if needed.
 func (t *logicTest) setUser(user string) {
+	if t.db != nil {
+		var dbName string
+
+		if err := t.db.QueryRow("SHOW DATABASE").Scan(&dbName); err != nil {
+			t.Fatal(err)
+		}
+
+		defer func() {
+			if _, err := t.db.Exec(fmt.Sprintf("SET DATABASE = %s", dbName)); err != nil {
```

Comment:
I believe you should be able to add it to the Open() call below using database=dbName and it should work